src/include/execution/vm/bytecode_handlers.h

```diff
@@ -1321,215 +1323,202 @@ VM_OP void OpIndexIteratorPerformInit(terrier::execution::sql::IndexIterator *it
 
 VM_OP_HOT void OpIndexIteratorScanKey(terrier::execution::sql::IndexIterator *iter) { iter->ScanKey(); }
 
-VM_OP_HOT void OpIndexIteratorAdvance(bool *has_more, terrier::execution::sql::IndexIterator *iter) {
-  *has_more = iter->Advance();
-}
+VM_OP_HOT void OpIndexIteratorScanAscending(terrier::execution::sql::IndexIterator *iter) { iter->ScanAscending(); }
```

Comment:
Are these really hot? They're implementations are not inlined, implying that they're not hot. Either (1) move `Scan*()` implementations to header or (2) mark these are WARM or move into `cpp`.

I'd suggest (2) for now and inline as we observe performance issues.